### PR TITLE
Use -rdynamic when building test executables in order to improve backtraces

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/flutter/issues/83286

### CODE REVIEW

File: testing/BUILD.gn [hunk 1]
```diff
@@ -6,6 +6,10 @@ import("//flutter/common/config.gni")
 import("//flutter/shell/config.gni")
 import("testing.gni")
 
+config("dynamic_symbols") {
+  ldflags = [ "-rdynamic" ]
```
Comment (chinmaygarde) [inline testing/BUILD.gn:10]: `if (is_clang && is_linux)` as this does not make sense on Windows.

File: testing/BUILD.gn [hunk 2]
```diff
@@ -37,6 +41,7 @@ source_set("testing") {
   ]
 
   public_deps = [ ":testing_lib" ]
+  public_configs = [ ":dynamic_symbols" ]
```
Comment (chinmaygarde) [inline testing/BUILD.gn:44]: ldflags don't need to go in public configs to be inherited by the `executable` or `shared_library` target. Source sets are not linked so `ldflags` are implicitly set on the final link step.

File: testing/BUILD.gn (hunk 1 above)
Comment (jason-simmons) [reply testing/BUILD.gn]: done

File: testing/BUILD.gn (hunk 2 above)
Comment (chinmaygarde) [reply testing/BUILD.gn:46]: I was wrong about this. `ldflags`, unlike `libs` on `source_sets` are silently ignored :(

**Review by chinmaygarde — APPROVED** [review]

**Comment by zanderso** [comment]
Landing on red to kick the tree following a timeout.

**Pull request merged**